src/Codec/Xlsx/Writer/Stream.hs

```diff
@@ -284,47 +293,44 @@ setNameSpaceRec space xelm =
                                     y -> y
     }
 
-columns :: Monad m => WriteSettings -> ConduitT SheetItem Event m ()
-columns settings =
+columns :: MonadReader WriteSettings m => ConduitT SheetItem Event m ()
+columns = do
+  colProps <- view wsColumnProperties
+  let cols :: Maybe TXML.Element
+      cols = nonEmptyElListSimple (n_ "cols") $ map (toElement (n_ "col")) colProps
   traverse_ (C.yieldMany . elementToEvents . toXMLElement) cols
-  where
-    cols :: Maybe TXML.Element
-    cols = nonEmptyElListSimple (n_ "cols") . map (toElement (n_ "col")) $ settings ^. wsColumnProperties
-
-writeWorkSheet :: Monad m => WriteSettings -> Map Text Int  -> ConduitT SheetItem Event m ()
-writeWorkSheet settings sstable = doc (n_ "worksheet") $ do
-    sheetViews settings
-    columns settings
-    el (n_ "sheetData") $ C.concatMap (mapItem sstable)
-
-
-mapItem :: Map Text Int -> SheetItem -> [Event]
-mapItem sstable sheetItem =
-  [EventBeginElement (n_ "row")  [("r", [ContentText $ toAttrVal rowIx])]]
-   <>
-  (ifoldMap (mapCell sstable rowIx) $ sheetItem ^. si_cell_row)
-   <>
-  [EventEndElement (n_ "row")]
+
+writeWorkSheet :: MonadReader WriteSettings  m => Map Text Int  -> ConduitT SheetItem Event m ()
```

Comment:
The refactor changes this type to some monad to a monadwriter, which allows us to make the settings available through all underlying code (pretty much all functions need some setting or another, explicitly telling which needs what gets tedious).